src/Lazy.js

```diff
@@ -2,16 +2,17 @@ import isSchema from './util/isSchema';
 
 class Lazy {
   constructor(mapFn) {
-    this._resolve = (...args) => {
-      let schema = mapFn(...args);
+    this._resolve = (value, options) => {
+      let schema = mapFn(value, options);
+
       if (!isSchema(schema))
         throw new TypeError('lazy() functions must return a valid schema');
 
-      return schema;
+      return schema.resolve(options);
     };
   }
-  resolve({ value, ...rest }) {
-    return this._resolve(value, rest);
+  resolve(options) {
+    return this._resolve(options.value, options);
```

Comment:
To allow nested `Lazy` or `when` to work we need not remove anything from `options`, so that you can `schema.resolve(options)` on returned schema from `lazy` function.

Reply:
hmm yeah, that's not great, it also means that every schema needs to now treat `options` as dynamic object bag instead of a defined shape that applies to all schema...it's fine for now. I may rethink how this is all structured, there are to many divergent paths between `lazy` and `when`